### PR TITLE
User defined column name in tally/count

### PR DESCRIPTION
**Pull request by quartin**
> There is currently no way to control the output variable name.

Following the premise that it would indeed be a good idea to add this functionality, I decided to give it a try, changing as little as possible the current functions. 
I've added a few tests, but maybe I'm missing some important corner cases. 

Feel free to disregard this if there's some reason not to do this (right now) or you have a more appropriate way in mind. I've also noticed there's been recent changes to the count/tally behavior in recent commits, so maybe this is not the best time.

### Example:
```
> df <- data_frame(name = c("a", "b", "a"), money = c(10, 20, 100))
> count(df, name, name = "n_accounts")
# A tibble: 2 x 2
  name  n_accounts
  <chr>      <int>
1 a              2
2 b              1
```

Regarding using an **already existing column name**, I'm not sure this is the appropriate solution, but I kept it equal to the behavior with `n`, adding a message:
```
> df <- data_frame(name = c("a", "b", "a"), money = c(10, 20, 100))
> count(df, name, name = "name")
Column `name` already exists, using `nname` as output variable name
# A tibble: 2 x 2
  name  nname
  <chr> <int>
1 a         2
2 b         1
```

The behavior of `add_count` and `add_tally` is exactly the same.

### CODE REVIEW

**Comment by romainfrancois** [comment]
Related to #3838

**Comment by hadley** [comment]
I have mixed feelings about this. On the one hand, I worry about `count()`/`tally()` getting increasingly complicated; I'd rather steer people towards using `summarise()` directly. OTOH this is simple change and adds little complexity.

But overall, I think this is a useful addition so we should incorporate it. @romainfrancois can you please do a review of the code?

**Comment by romainfrancois** [comment]
Implementation looks good. If we allow the possibility of giving the name, why not just `stop()` if the supplied name is a column name ?

File: R/count-tally.R [hunk 1]
```diff
@@ -151,7 +151,12 @@ add_tally <- function(x, wt, sort = FALSE) {
     n <- quo(sum(!!wt, na.rm = TRUE))
   }
 
-  n_name <- n_name(tbl_vars(x))
+  n_name <- n_name(tbl_vars(x), name)
```
Comment (romainfrancois) [inline R/count-tally.R]:
```suggestion
  n_name <- n_name(group_vars(x), name)
```

same as above, would deal with #3838

**Comment by quartin** [comment]
> I'm still wondering if instead of keep adding n until it's fine, perhaps we should error if the name is in the group_vars()

Should this apply to every name, including `n`? I tend to agree (always found it an odd behavior), but it will be a breaking change.

**Comment by romainfrancois** [comment]
Merging this now, we can think later if it's worth aborting instead of add the `n` prefix until.
